how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts

```diff
@@ -0,0 +1,217 @@
+	const loggingContainer: HTMLDivElement | null = document.querySelector("#logging-container");
+
+	if (!loggingElement || !loggingContainer) {
+		return;
+	}
+
+	loggingContainer.style.display = "flex";	
+
+	clearLogsBtn = document.querySelector("#btnClear");
+	clearLogsBtn?.addEventListener("click", () => loggingElement ? loggingElement.textContent = "" : null);
+
+	connectToSideCarBtn = document.querySelector("#btnConnectToSideCarApp");
+	connectToSideCarBtn?.addEventListener("click", async ()=> {
+		await setupSideCarApp();
+		if(connectToSideCarBtn !== null) {
+			connectToSideCarBtn.disabled = true;
+		}
+		if(dispatchMessageToSideCarBtn !== null) {
+			dispatchMessageToSideCarBtn.disabled = false;
+		}
+	});
+
+	dispatchMessageToSideCarBtn = document.querySelector("#btnDispatchMessageToSideCarApp");
+	dispatchMessageToSideCarBtn?.addEventListener("click", dispatchMessageToSideCarApp);
+}
+
+/**
+ * Initialize the Platform.
+ */
+async function initializePlatform(): Promise<void> {
+	try {
+		await fin.Platform.init({ });
+		loggingAddEntry("Platform initialized.");
+	} catch(error) {
+		loggingAddEntry(`There was an error while trying to initialize your platform. Error: ${getErrorMessage(error)}`);		
+	}
+}
+
+/**
+ * Return a message from an error.
+ * @param error The error to inspect and retrieve the message from
+ * @returns The error message
+ */
+function getErrorMessage(error: unknown): string {
+	if (error instanceof Error) return error.message
+	return String(error)
+}
+
+/**
+ * Create and assign a channel client
+ * @returns channel client
+ */
+async function createChannelClient(): Promise<ChannelClient> {
+	const channelName = "sidecar-app";
+    const channelClient = await fin.InterApplicationBus.Channel.connect(channelName, { payload: "Payload from platform"});
+	loggingAddEntry(`Connection To SideCar App on channel: ${channelName} established.`);
+	return channelClient;
+}
+
+/**
+ * Fetch the SideCar App, Launch it and Connect to it via Channels.
+ */
+async function setupSideCarApp() {
+	await fetchSideCarAppIfNeeded();
+	await launchSideCarApp();
+	await createChannelClientAndRegisterListeners();
+}
+
+/**
+ * Create a connection to the SideCar application.
+ */
+async function createChannelClientAndRegisterListeners(): Promise<void> {
+	channelClient = await createChannelClient();
+
+	if(channelClient !== null) {
+		channelClient.onDisconnection((identity) => {
+			console.log("onDisconnection identity: ", identity);
+			loggingAddEntry(`onDisconnection identity: ${JSON.stringify(identity)}`);
+			if(connectToSideCarBtn !== null) {
+				connectToSideCarBtn.disabled = false;
+			}
+			if(dispatchMessageToSideCarBtn !== null) {
+				dispatchMessageToSideCarBtn.disabled = true;
+			}
+			loggingAddEntry("Client has been disconnected from the SideCar App");
+		});
+
+		// received request over channel to display a message
+		channelClient.register('sidecar-app-client-subscriber', async (payload) => {
```

Comment:
Same with naming here, can we make global const

Reply:
done